src/main/resources/assets/js/components/AppComponent.jsx

```diff
@@ -4,8 +4,16 @@ var React = require("react/addons");
 var AppHealthComponent = require("../components/AppHealthComponent");
 
 var AppComponent = React.createClass({
+  propTypes: {
+    model: React.PropTypes.object.isRequired,
+    router: React.PropTypes.object.isRequired
+  },
+
   onClick: function () {
-    this.props.onClick(this.props.model);
+    this.props.router.navigate(
```

Comment:
I agree. When we do get the time, we should change to Flux structure. We are already using it in other projects :)